cloudwatch/cloudwatch.go

```diff
@@ -289,10 +313,20 @@ func (output *OutputPlugin) AddEvent(e *Event) int {
 		return fluentbit.FLB_OK
 	}
 
-	// Step 3. Assign a log group and log stream name to the Event.
+	// Step 3. Extract the Task Metadata if applicable.
+	if output.runningInECS && output.ecsMetadata.TaskID == "" {
+		err := output.getECSMetadata()
+		if err != nil {
+			logrus.Errorf("[cloudwatch %d] Failed to get ECS Task Metadata with error: %v\n", output.PluginInstanceID, err)
```

Comment:
You end up logging the error twice with this line. See next comment for suggestion.